CODING_STANDARDS.md

```diff
@@ -15,7 +15,18 @@ We adhere to the `pep8` standard, using `flake8`, `pylint` and `mypy` to catch f
 
 ## Solidity
 
-As of now, the solidity code does not follow consistent style conventions. Nevertheless, new solidity code must adhere to the [solidity coding standards](https://docs.soliditylang.org/en/develop/style-guide.html). Old code will progressively be updated to converge towards the recommended style.
+Solidity code must adhere to the [solidity coding standards](https://docs.soliditylang.org/en/develop/style-guide.html), with the following amendments:
+- Test functions must end with the suffix `Test` (e.g. `reverseBytesTest`)
+- Private/internal state variables and functions must have an underscore prefix (e.g. `_myInternalFunction`)
+- Virtual functions in abstract contracts must be written last and located at the end of the contract source
+- Function parameters must not be prefixed with an underscore
+- Interface names must have a capital I prefix (e.g. `IERC20`)
```

Comment:
@dtebbs - by any chance, do you remember why `IERC20` is declared as an abstract contract rather than an interface? 
See: https://github.com/clearmatics/zeth/blob/adjust-solidity-standards/zeth_contracts/contracts/Tokens.sol#L11

Reply:
> do you remember why IERC20 is declared as an abstract contract rather than an interface?

TBH I haven't considered this, but IIRC `interface` was not available in old versions of Solidity.  My guess is that that is part of the reason.

Reply:
Hum... Interfaces are supported since a pretty long time: https://github.com/ethereum/solidity/blob/develop/Changelog.md#040-2016-09-08 - not sure.. anyway, will switch to an interface.